### PR TITLE
Turkish braille table grade 2

### LINKED ISSUE #16735: Request for Turkish Grade II table
### Is your feature request related to a problem? Please describe.

NVDA 2024.2 includes Liblouis 3.29.  This version of LibLouis includes a Turkish Grade II table, however this has not been enabled in NVDA.

### Describe the solution you'd like

Enable the Turkish Grade II table for Braille users of NVDA.

### Additional context
(Requested by email as "Suggestion request")

### PR DESCRIPTION
**Pull request by zstanecic**
### Link to issue number:
Fixes #16735
### Summary of the issue:
This adds Turkish grade 2 braille table to NVDA interface.
### Description of user facing changes
Users will be able to use the table provided in NVDA
### Description of development approach
Modified the brailleTables.py.
### Testing strategy:
Tested from source with the braille display that the braille table works
### Known issues with pull request:
None known
### Code Review Checklist:

- [x ] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x ] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x ] API is compatible with existing add-ons.
- [x ] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Added support for Turkish grade 2 Braille table with contracted Braille.
  - Added support for Setswana grade 1 Braille table.

- **Documentation**
  - Updated user documentation to reflect the new Turkish grade 2 Braille table.
  - Other documentation updates include support for the Karakalpak language and various version updates.

- **Updates**
  - Updated Braille tables for several languages and scripts.
  - Upgraded dependencies including LibLouis, eSpeak, fast_diff_match_patch, NSIS, Python Markdown, and nh3 to their latest versions.

### CODE REVIEW

**Comment by zstanecic** [comment]
If something needs to be corrected here, let me anybody know.

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Unicode CLDR has also been updated.
   * Updated fast_diff_match_patch (used to detect changes in terminals and other dynamic content) to version 2.1.0. (#16508, @codeofdusk)
   * Updated LibLouis Braille translator to [3.30.0](https://github.com/liblouis/liblouis/releases/tag/v3.30.0). (#16652, @codeofdusk)
     * New Braille tables for Cyrillic Serbian, Yiddish, several ancient languages (Biblical Hebrew, Akkadian, Syriac, and Ugaritic), and transliterated Cuneiform text.
+    * New braille table for turkish grade2. (#16735)
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply user_docs/en/changes.md:42]:
```suggestion
    * New braille table for Turkish grade 2. (#16735)
```

**Review by seanbudd — APPROVED** [review]

**Comment by oguzugur28** [comment]
Hi all,
To  adding Turkish Grade II Braille table to NVDA was requested by me via email.
I'm appriciates to your excellent work.
I can check fix related about Turkish Grade II Braille table, but for now, I don't know that how i can check it? My question may be so easy but unfortunately I'm not a software expert. If you can show to test ways, I would be test it.
Best,

**Comment by zstanecic** [comment]
Hi Oguz,

You can install the latest alpha snapshot, or run it in a portable copy, 
and test it there if you have a braille display.

Best.

Zvonimir

**Comment by oguzugur28** [comment]
Hi Zvonimir,
Thanks for your answer.
I've braille display. Where I can find the latest alpha snapshot?
Best,

**Comment by seanbudd** [comment]
@oguzugur28 - https://www.nvaccess.org/files/nvda/snapshots/

**Pull request merged**